src/EnergyPlus/HeatBalanceSurfaceManager.cc

```diff
@@ -4877,6 +4906,311 @@ namespace HeatBalanceSurfaceManager {
+                                    Zone(ZoneNum).Name);
+            }
+            lowSETLongestHours.assign(NumOfZones, 0.0);
+            highSETLongestHours.assign(NumOfZones, 0.0);
+            lowSETLongestStart.assign(NumOfZones, 0.0);
+            highSETLongestStart.assign(NumOfZones, 0.0);
+            reportThermalResilienceFirstTime = false;
+        }
+
+        // Calculate Heat Index and Humidex.
+        // The heat index equation set is fit to Fahrenheit units, so the zone air temperature values are first convert to F,
+        // then heat index is calculated and converted back to C.
+        for (int ZoneNum = 1; ZoneNum <= NumOfZones; ++ZoneNum) {
+            Real64 ZoneT = ZTAV(ZoneNum);
+            Real64 ZoneW = ZoneAirHumRatAvg(ZoneNum);
+            Real64 ZoneRH = Psychrometrics::PsyRhFnTdbWPb(ZoneT, ZoneW, OutBaroPress) * 100.0;
+            Real64 ZoneTF = ZoneT * (9.0 / 5.0) + 32.0;
+            Real64 HI;
+
+            if (ZoneTF < 80) {
+                HI = 0.5 * (ZoneTF + 61.0 + (ZoneTF - 68.0) * 1.2 + (ZoneRH * 0.094));
+            } else {
+                HI = c1 + c2 * ZoneTF + c3 * ZoneRH + c4 * ZoneTF * ZoneRH + c5 * ZoneTF * ZoneTF + c6 * ZoneRH * ZoneRH +
+                     c7 * ZoneTF * ZoneTF * ZoneRH + c8 * ZoneTF * ZoneRH * ZoneRH + c9 * ZoneTF * ZoneTF * ZoneRH * ZoneRH;
+                if (ZoneRH < 13 && ZoneTF < 112) {
+                    HI -= (13 - ZoneRH) / 4 * std::sqrt((17 - abs(ZoneTF - 95)) / 17);
+                } else if (ZoneRH > 85 && ZoneTF < 87) {
+                    HI += (ZoneRH - 85) / 10 * (87 - ZoneTF) / 5;
+                }
+            }
+
+            HI = (HI - 32.0) * (5.0 / 9.0);
+
+            Real64 TDewPointK = Psychrometrics::PsyTdpFnWPb(ZoneW, OutBaroPress) + KelvinConv;
+            Real64 e = 6.11 * std::exp(5417.7530 * ((1 / 273.16) - (1 / TDewPointK)));
+            Real64 h = 5.0 / 9.0 * (e - 10.0);
+            Real64 Humidex = ZoneT + h;
+
+            ZoneHeatIndex(ZoneNum) = HI;
+            ZoneHumidex(ZoneNum) = Humidex;
+        }
+
+        // Count hours only during weather simulation periods
+        if (ksRunPeriodWeather == KindOfSim && !WarmupFlag) {
+            // Trace current time step Zone Pierce SET; NaN if no occupant or SET not calculated
+            // Record last time step SET to trace SET unmet duration;
+            for (int iPeople = 1; iPeople <= TotPeople; ++iPeople) {
+                int ZoneNum = People(iPeople).ZonePtr;
+                ZoneNumOcc(ZoneNum) = People(iPeople).NumberOfPeople * GetCurrentScheduleValue(People(iPeople).NumberOfPeoplePtr);
+                ZoneOccPierceSETLastStep(ZoneNum) = ZoneOccPierceSET(ZoneNum);
+                if (ZoneNumOcc(ZoneNum) > 0) {
+                    if (People(iPeople).Pierce) {
+                        ZoneOccPierceSET(ZoneNum) = ThermalComfort::ThermalComfortData(iPeople).PierceSET;
+                    } else {
+                        ZoneOccPierceSET(ZoneNum) = NAN;
```

Comment:
@mjwitte This is to explicitly set the value to some kind of "null" instead of zero during runtime at certain time steps (no occupant). If that is not working, it would work the same to set it to some unreasonable values like "999" or "-999" for temperatures or add more logics with flags like `ZoneOccPierceSETIsNull" or `ZoneOccPierceSETLastTimeStepIsNull".

Reply:
-999 is fine.  Or just -1 if that can be used as a sentinel?